raft/src/main/java/org/apache/kafka/raft/internals/RecordsBatchReader.java

```diff
@@ -117,14 +117,8 @@ private void ensureOpen() {
     }
 
     private Optional<Batch<T>> nextBatch() {
-        while (iterator.hasNext()) {
-            Batch<T> batch = iterator.next();
-
-            if (batch.records().isEmpty()) {
-                lastReturnedOffset = batch.lastOffset();
-            } else {
-                return Optional.of(batch);
-            }
+        if (iterator.hasNext()) {
```

Comment:
We're testing this change indirectly, but perhaps it is useful to have an explicit test in `RecordsBatchReaderTest` as well.

Reply:
I agree. I added a couple of tests for this.